packages/grid/x-data-grid/src/hooks/features/pagination/useGridPagination.ts

```diff
@@ -44,17 +81,212 @@ export const useGridPagination = (
+    | 'paginationModel'
+    | 'onPaginationModelChange'
     | 'autoPageSize'
     | 'rowCount'
     | 'initialState'
     | 'paginationMode'
+    | 'signature'
     | 'rowHeight'
   >,
 ) => {
-  useGridPageSize(apiRef, props);
-  useGridPage(apiRef, props);
+  const logger = useGridLogger(apiRef, 'useGridPagination');
+
+  const visibleTopLevelRowCount = useGridSelector(apiRef, gridVisibleTopLevelRowCountSelector);
+  const densityFactor = useGridSelector(apiRef, gridDensityFactorSelector);
+  const rowHeight = Math.floor(props.rowHeight * densityFactor);
+
+  apiRef.current.registerControlState({
+    stateId: 'pagination',
+    propModel: props.paginationModel,
+    propOnChange: props.onPaginationModelChange,
+    stateSelector: gridPaginationModelSelector,
+    changeEvent: 'paginationModelChange',
+  });
+
+  /**
+   * API METHODS
+   */
+  const setPage = React.useCallback<GridPaginationApi['setPage']>(
+    (page) => {
+      const currentModel = gridPaginationModelSelector(apiRef);
+      if (page === currentModel.page) {
+        return;
+      }
+      logger.debug(`Setting page to ${page}`);
+      apiRef.current.setPaginationModel({ page, pageSize: currentModel.pageSize });
+    },
+    [apiRef, logger],
+  );
+
+  const setPageSize = React.useCallback<GridPaginationApi['setPageSize']>(
+    (pageSize) => {
+      const currentModel = gridPaginationModelSelector(apiRef);
+      if (pageSize === currentModel.pageSize) {
+        return;
+      }
+
+      logger.debug(`Setting page size to ${pageSize}`);
+      apiRef.current.setPaginationModel({ pageSize, page: currentModel.page });
+    },
+    [apiRef, logger],
+  );
+
+  const setPaginationModel = React.useCallback<GridPaginationApi['setPaginationModel']>(
+    (paginationModel) => {
+      const currentModel = gridPaginationModelSelector(apiRef);
+      if (paginationModel === currentModel) {
+        return;
+      }
+      logger.debug("Setting 'paginationModel' to", paginationModel);
+
+      apiRef.current.updateControlState(
+        'pagination',
+        mergeStateWithPaginationModel(
+          props.rowCount ?? visibleTopLevelRowCount,
+          props.autoPageSize,
+          props.signature,
+          paginationModel,
+        ),
+        'setPaginationModel',
+      );
+      apiRef.current.forceUpdate();
+    },
+    [apiRef, logger, props.autoPageSize, props.rowCount, props.signature, visibleTopLevelRowCount],
+  );
+
+  const pageApi: GridPaginationApi = {
+    setPage,
+    setPageSize,
+    setPaginationModel,
+  };
+
+  useGridApiMethod(apiRef, pageApi, 'public');
+
+  /**
+   * PRE-PROCESSING
+   */
+  const stateExportPreProcessing = React.useCallback<GridPipeProcessor<'exportState'>>(
+    (prevState, context) => {
+      const paginationModel = gridPaginationModelSelector(apiRef);
+
+      const shouldExportPaginationModel =
+        // Always export if the `exportOnlyDirtyModels` property is activated
+        !context.exportOnlyDirtyModels ||
```

Comment:
@flaviendelangle should the comment be

```suggestion
        // Always export if the `exportOnlyDirtyModels` property is not activated
        !context.exportOnlyDirtyModels ||
```

Reply:
:see_no_evil: yes
And the comment error is probably present in all plugin